### PR TITLE
[UWP] Use a shared output queue for sw av1 and vpx decoder

### PR DESCRIPTION
**Pull request by victorpasoshnikov**
Reorganize output queue. Use one large preallocated hw buffer for both av1 & vpx sw decoders

b/249739051

Change-Id: I6430ae1ba5d288ed2495f3056cc7283f5c189f49

### CODE REVIEW

File: starboard/shared/uwp/extended_resources_manager.cc
Comment (TyHolc) [inline starboard/shared/uwp/extended_resources_manager.cc:63]: Typo for 'underlying'

File: starboard/shared/uwp/extended_resources_manager.cc
Comment (TyHolc) [inline starboard/shared/uwp/extended_resources_manager.cc:275]: Might want to add a SB_DCHECK after this if-block to keep consistent with d3d12device_ and d3d12queue_

**Review by TyHolc — COMMENTED** [review]
Added a couple comments but overall looks good to me. I'll leave it up to @osagie98 to review the media changes.

File: starboard/shared/uwp/extended_resources_manager.cc
Comment (victorpasoshnikov) [reply starboard/shared/uwp/extended_resources_manager.cc:63]: Done

File: starboard/shared/uwp/extended_resources_manager.cc
Comment (victorpasoshnikov) [reply starboard/shared/uwp/extended_resources_manager.cc:275]: Done

**Review by osagie98 — APPROVED** [review]
LGTM as long as this doesn't create unnecessary memory pressure. I have a few small comments.

File: starboard/shared/uwp/extended_resources_manager.cc
Comment (osagie98) [inline starboard/shared/uwp/extended_resources_manager.cc:500]: Change to SB_LOG so it will log in QA builds

File: starboard/shared/uwp/extended_resources_manager.h
Comment (osagie98) [inline starboard/shared/uwp/extended_resources_manager.h:50]: Add to this comment to include the buffer heap

File: starboard/shared/uwp/extended_resources_manager.cc
Comment (victorpasoshnikov) [reply starboard/shared/uwp/extended_resources_manager.cc]: Done

File: starboard/shared/uwp/extended_resources_manager.h
Comment (victorpasoshnikov) [reply starboard/shared/uwp/extended_resources_manager.h]: Done

**Comment by victorpasoshnikov** [comment]
The internal part is reviewed by @jasonzhangxx

**Review by jasonzhangxx — APPROVED** [review]

**Pull request merged**
